server.js

```diff
@@ -0,0 +1,188 @@
+// DEPENDENCIES
+const express = require('express');
+const logger = require('morgan');
+const cors = require('cors');
+const helmet = require('helmet');
+
+// DATABASE HELPERS
+const userDb = require('./data/helpers/userDb');
+const postDb = require('./data/helpers/postDb');
+const tagDb = require('./data/helpers/tagDb');
+
+// SERVER
+const server = express();
+
+// MIDDLEWARE
+const cruiseControl = (req, res, next) => {
+	req.body.name = req.body.name.toUpperCase();
+	next();
+};
+
+server.use(logger('tiny'), cors(), helmet(), express.json());
+
+// ENDPOINTS
+// get users
+server.get('/api/users', (req, res) => {
+	userDb
+		.get()
+		.then(users => {
+			console.log('\n=== RETURNED USERS: ===\n\n', users, '\n');
+			res.status(200).json(users);
+		})
+		.catch(err => {
+			console.log('\n=== UH OH ===\n', err);
+			res
+				.status(500)
+				.json({ error: 'Users information could not be retrieved.' });
+		});
+});
+
+// get user by id
+server.get('/api/users/:id', (req, res) => {
+	userDb
+		.get(req.params.id)
+		.then(user => {
+			if (!user) {
+				console.log('\n=== NO USER BY THAT ID ===\n');
+				return res
+					.status(404)
+					.json({ message: 'User with specified ID does not exist.' });
+			}
+			console.log('\n=== RETURNED USER: ===\n\n', user, '\n');
+			res.status(200).json(user);
+		})
+		.catch(err => {
+			console.log('\n=== UH OH ===\n\n', err);
+			res
+				.status(500)
+				.json({ error: 'User information could not be retrieved.' });
+		});
+});
+
+// add new user
+//    todo: return message when name already exists
+server.post('/api/users', cruiseControl, (req, res) => {
+	if (!req.body.name) {
+		console.log(`\n=== USER NAME NOT PROVIDED ===\n\n`);
+		return res.status(400).json({
+			errorMessage: 'Please provide name for the user.'
+		});
+	}
+
+	if (req.body.name.length > 128) {
+		console.log(`\n=== USER NAME EXCEEDED 128 CHARACTER LIMIT ===\n\n`);
+		return res.status(400).json({
+			errorMessage: 'User name cannot exceed 128 characters.'
+		});
+	}
+
+	userDb
+		.insert(req.body)
+		.then(({ id }) => {
+			userDb.getUser(id).then(user => {
```

Comment:
I am able to post successfully, but getting an error here that getUser is not a function. Did you maybe mean to do just `get` here and supply the id? Note you can check what helper methods are available in `data/helpers/userDb`.